federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/csw/probe/DiscoverCswSourceProbeMethod.java

```diff
@@ -13,131 +13,77 @@
+import static org.codice.ddf.admin.commons.sources.SourceHandlerCommons.DISCOVERED_URL;
+import static org.codice.ddf.admin.commons.sources.SourceHandlerCommons.UNKNOWN_ENDPOINT;
+import static org.codice.ddf.admin.commons.sources.SourceHandlerCommons.getCommonSourceSubtypeDescriptions;
+import static org.codice.ddf.admin.sources.csw.CswSourceUtils.discoverCswUrl;
+import static org.codice.ddf.admin.sources.csw.CswSourceUtils.getPreferredCswConfig;
+import static org.codice.ddf.admin.commons.sources.SourceHandlerCommons.DISCOVER_SOURCES_ID;
 
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
 
 import org.codice.ddf.admin.api.config.sources.CswSourceConfiguration;
 import org.codice.ddf.admin.api.handler.ConfigurationMessage;
-import org.codice.ddf.admin.api.handler.commons.UrlAvailability;
 import org.codice.ddf.admin.api.handler.method.ProbeMethod;
 import org.codice.ddf.admin.api.handler.report.ProbeReport;
-import org.codice.ddf.admin.sources.csw.CswSourceUtils;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
 
 public class DiscoverCswSourceProbeMethod extends ProbeMethod<CswSourceConfiguration> {
 
     public static final String CSW_DISCOVER_SOURCES_ID = DISCOVER_SOURCES_ID;
-
-    public static final String DESCRIPTION =
-            "Attempts to discover a CSW endpoint based on a hostname and port using optional authentication information.";
-
-    public static final List<String> REQUIRED_FIELDS = ImmutableList.of(SOURCE_HOSTNAME, PORT);
-
-    public static final List<String> OPTIONAL_FIELDS = ImmutableList.of(SOURCE_USERNAME,
-            SOURCE_USER_PASSWORD);
-
-    public static final Map<String, String> SUCCESS_TYPES = ImmutableMap.of(CONFIG_CREATED,
-            "Successfully created a configuration from the CSW endpoint.");
-
-    public static final Map<String, String> FAILURE_TYPES = ImmutableMap.of(CANNOT_CONNECT,
-            "The URL provided could not be reached.",
-            UNKNOWN_ENDPOINT,
-            "The endpoint does not appear to have CSW capabilities.",
-            CERT_ERROR,
-            "The discovered source has incorrectly configured SSL certificates and is insecure.",
-            INTERNAL_ERROR,
-            "Failed to create configuration from CSW URL.");
-
-    public static final Map<String, String> WARNING_TYPES = ImmutableMap.of(UNTRUSTED_CA,
-            "The discovered URL has incorrectly configured SSL certificates and is likely insecure.");
-
+    public static final String DESCRIPTION = "Attempts to discover a CSW endpoint based on a hostname and port using optional authentication information. If the \"endpointUrl\" is specified it will use this to create a configuration instead of discovering the url.";
+    public static final List<String> OPTIONAL_FIELDS = ImmutableList.of(SOURCE_HOSTNAME, PORT, ENDPOINT_URL, SOURCE_USERNAME, SOURCE_USER_PASSWORD, ENDPOINT_URL);
```

Comment:
We're now calling this method with either `hostname` and `port` or an `endpointUrl`. Technically they're all optional, but you have to have one or the other. The validation is checking that we do have one or the other (`validateOptionalFields` below)